hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -630,8 +642,26 @@ def generate_new_examples(self):
 
         self.health_check_state = HealthCheckState()
 
+        def should_generate_more():
+            if not self.interesting_examples:
+                return True
+            assert self.first_bug_calls <= self.last_bug_calls <= self.call_count
+            return (
+                # Shrinking must be enabled as continued generation may stumble
+                # upon smaller examples of previously-found bugs.
+                Phase.shrink in self.settings.phases
+                # We cap 'calls after first bug' so errors are reported reasonably
+                # quickly even for tests that may run for an extremely long time.
+                and self.call_count < self.first_bug_calls + 1000
+                # Finally, we try at least ten calls because the simplest possible
+                # input often triggers a unique bug, and if there is another trivial
+                # bug we'd like to find that too.  Otherwise, we stop after making
+                # as many more calls as it took to find the most recent bug.
+                and self.call_count < max(10, self.last_bug_calls * 2)
```

Comment:
Just to clarify, the intent here seems to be “keep going until the most recent 50% of our search effort has found no novel bugs”.

Is that accurate?

Reply:
Yes, that's correct, with a minimum search effort of ten calls.